core/trino-main/src/main/java/io/trino/server/protocol/Query.java

```diff
@@ -545,14 +545,14 @@ private synchronized void closeExchangeClientIfNecessary(QueryInfo queryInfo)
         }
     }
 
-    private void handleSerializationException(Throwable exception)
+    private synchronized void handleSerializationException(Throwable exception)
     {
         // failQuery can throw exception if query has already finished.
         try {
             queryManager.failQuery(queryId, exception);
         }
         catch (RuntimeException e) {
-            log.debug("Could not fail query", e);
```

Comment:
we should have static analysis to catch that.

Reply:
IntelliJ found this one. I created https://github.com/airlift/airlift/pull/922 for use with https://errorprone.info/bugpattern/FormatStringAnnotation